### ISSUE TITLE
Compatibility with Cornerstone3D

### ISSUE THREAD

**Issue by kousu** [cornerstonejs/cornerstone-nifti-image-loader#48]
I want to use this but I don't know how I am supposed to right now. Can you clarify if this is/isn't compatible with the new Cornerstone3D? `package.json` doesn't list it as a dependency (peer or otherwise) so I'm not sure what version of Cornerstone I'm supposed to be using.

According to https://www.cornerstonejs.org/, Cornerstone3D, or [`@cornerstonejs/core`](https://www.npmjs.com/package/@cornerstonejs/core), is the new "beta" version of Cornerstone:

![Screenshot_20221006_175839](https://user-images.githubusercontent.com/987487/194429236-3849579f-ef9c-46d5-84c4-9f65fa2c8294.png)

and according to https://github.com/cornerstonejs/cornerstone3D-beta/issues/118#issuecomment-1270731312, the legacy Cornerstone, [`cornerstone-core`](https://www.npmjs.com/package/cornerstone-core), is no longer being maintained at all.

The two projects have separate versioning schemes. The legacy one is at 2.6.1, the beta is at 0.16.7. So they really are separate projects.

The reason I'm confused is:

- this repo is installable as [`@cornerstonejs/nifiti-image-loader`](https://www.npmjs.com/package/@cornerstonejs/nifti-image-loader), as if it's a peer of `@cornerstonejs/core`, which has the most recent 1.0.7
- but it's *also* installable as the older name [`cornerstone-nifiti-image-loader`](https://www.npmjs.com/package/cornerstone-nifti-image-loader), though that copy hasn't been updated since 1.0.0
- the actual examples here have _vendored_ a copy of the legacy code, and haven't touched it in 5 years

    https://github.com/cornerstonejs/cornerstone-nifti-image-loader/blob/b14ae74537d56ea17a282350490949c77512d71f/examples/cornerstone.js#L1

So, how do I use this? If I `npm i @cornerstone/core` will this work correctly with it?

**Comment by kousu** [comment]
(I do plan to try to test and figure this out on my own. I'm posting this to ask for some guidance to be added to the docs to help others out during the transition)

**Comment by simonalo** [comment]
I'm currently trying to make this project work with cornerstone3D but there are functions which changed with corenerstone3D (as loadAndCacheImage for example). I don't know if the both are compatible.

**Comment by acegank** [comment]
i load nifti files for Cornerstone3D, but it not works.

are you have some examples?

**Comment by kousu** [comment]
> I'm currently trying to make this project work with cornerstone3D but there are functions which changed with corenerstone3D (as loadAndCacheImage for example). I don't know if the both are compatible.

Thank you for the clarification! That helps a lot. I'll build on [`cornerstone-core^2.6.1`](https://www.npmjs.com/package/cornerstone-core) + [`cornerstore-nifti-image-loader`](https://www.npmjs.com/package/cornerstone-nifti-image-loader) then. I feel pretty stupid for not being able to figure that out from [the docs](https://www.cornerstonejs.org/docs/getting-started/overview), but they are pretty sprawling and I found myself getting lost in them. Thank you for cutting through the noise for me @simonalo.

**Comment by kousu** [comment]
@acegank there's an example [right in the repo](https://github.com/cornerstonejs/cornerstone-nifti-image-loader/blob/b14ae74537d56ea17a282350490949c77512d71f/examples/index.html).

But like I said, and like @simonalo confirmed, these NIfTI examples need to work against `cornerstone-core` otherwise known as legacy cornerstone; `@cornerstonejs/core` otherwise known as Cornerstone3D is not yet compatible with NIfTI files.

**Comment by MYWpro** [comment]
> load a NIFTI as a volume

Did you finish the job? I really don't understand the documentation, can I use  imagevolume  class to do this task directly?

**Comment by MYWpro** [comment]
> load a NIFTI as a volume

@scandel  
Did you finish the job? I really don't understand the documentation, can I use imagevolume class to do this task directly?

**Comment by pcanas** [comment]
Is this something planned to be tackled soon?

**Comment by Ladon90** [comment]
@scandel
Thank you very much for your contribution and the explanations!
Did you meanwhile found a way to load NIFTI into a volume?

Kind regards

**Comment by ShilpaP17** [comment]
Yes you can load NIFTI images in cornerstone3D in volume using @cornerstonejs/nifti-volume-loader.
I am able to load in react typescript project with webpack

**Comment by Ladon90** [comment]
@ShilpaP17 
Such great news, thanks a lot!

I directly tested it with vuetify3-vue3-vite, but unfortunately my volumes are shown pixelated:

![image](https://github.com/cornerstonejs/cornerstone-nifti-image-loader/assets/133389125/d3164672-dfaf-4895-abb6-f8818c1650c5)

Is there anyone with similar issues or a solution?

Thank you in advance
